--- a/src/lib/Proxy.py	2010-11-02 16:53:43.404016001 -0700
+++ b/src/lib/Proxy.py	2010-11-02 16:56:38.654016003 -0700
@@ -250,7 +250,7 @@
         self.timeout = timeout
 
     def make_connection(self, host):
-        host = self.get_host_info(host)[0]
+        host, self._extra_headers = self.get_host_info(host)[0:2]
         http = SSLHTTPConnection(host, key=self.key, cert=self.cert, ca=self.ca,
                                  scns=self.scns, timeout=self.timeout)
         https = httplib.HTTP()
