Index: debian/buildsys/common/bcfg2-server.init
===================================================================
--- debian/buildsys/common/bcfg2-server.init	(revision 2796)
+++ debian/buildsys/common/bcfg2-server.init	(working copy)
@@ -37,6 +37,7 @@
     if [ "$STATUS" = 0 ]
     then
 	log_success_msg "bcfg2-server"
+	touch /var/lock/subsys/bcfg2-server
     else
 	log_failure_msg "bcfg2-server"
     fi
@@ -48,6 +49,7 @@
     STATUS=$?
     if [ "$STATUS" = 0 ]; then
       log_success_msg "bcfg2-server"
+      rm -f /var/lock/subsys/bcfg2-server
       exit 0
     else
       log_failure_msg "bcfg2-server"
